pyls/__main__.py

```diff
@@ -95,3 +80,37 @@ def _binary_stdio():
         stdin, stdout = sys.stdin, sys.stdout
 
     return stdin, stdout
+
+def configure_logger(args):
+    log_config = args.log_config
+    root_logger = logging.root
+
+    if log_config:
+        with open(log_config, 'r') as f:
+            logging.config.dictConfig(json.load(f))
+    else:
+        log_file = args.log_file
+        formatter = logging.Formatter(LOG_FORMAT, style="%")
+        if log_file:
+            log_handler = logging.handlers.RotatingFileHandler(log_file, mode='a',
+                    maxBytes=50*1024*1024, backupCount=10, encoding=None, delay=0)
+        else:
+            log_handler = logging.StreamHandler()
+        log_handler.setFormatter(formatter)
+        root_logger.addHandler(log_handler)
+
+    # CRITICAL = 50
+    # FATAL = CRITICAL
+    # ERROR = 40
+    # WARNING = 30
+    # INFO = 20
+    # DEBUG = 10
```

Comment:
Why are these comments here? They don't seem to be relevant to the code

Reply:
Informative about the debug levels. I put it there because I was confused about them when I saw you can only pass 3 levels, but I know the `logging` module has 6. So, looking at these values I can clearly see that we you set the WARNING level, you can also see the ERROR and other above it.

Reply:
I think that's part of the logging API and doesn't need re-documenting here. Most logging frameworks would log at the given level or "above"